cluster.go

```diff
@@ -236,15 +252,27 @@ func (c *clusterClient) _refresh() (err error) {
 					pslots[i] = conns[g.nodes[1+util.FastRand(nodesCount-1)]].conn
 				}
 			}
-		case c.rOpt != nil: // implies c.opt.SendToReplicas != nil
+		case c.rOpt != nil:
 			if len(rslots) == 0 { // lazy init
 				rslots = make([]conn, 16384)
 			}
 			if len(g.nodes) > 1 {
+				n := len(g.nodes) - 1
+				replicas := make([]ReplicaInfo, 0, n)
+				for _, addr := range g.nodes[1:] {
```

Comment:
I already consider it. But i didn't. Because

1. `g.nodes` has primary node too. so `ReplicaInfo` slice type is incorrect. It can be misleading to contributors.
2. user can change value each `ReplicaInfo`. Changing `ReplicaInfo` is not problem to rueidis. but there can be side effect in the user side.

Because of duplicated allocation, i also considering reusing `ReplicaInfo` slice. But problem is If user holds `ReplicaInfo` slice in the `ReplicaSelector` it can be problem.

I prefer reusing `ReplicaInfo` slice if should reduce allocation. And in this case, we should add comment that user must not store reference `ReplicaInfo` slice.

Reply:
Hi @proost,

Great considerations!

1. If naming is the concern, we can use some sort of the type alias. We can only expose ReplicaInfo to users but use another name internally.
2. We can add comments on the selector function to tell the user should not change values in the slice.